docs/source/links/state.md

```diff
@@ -0,0 +1,609 @@
+to set these features up yourself. 🎉 On top of that, you also benefit from the
+excellent developer experience of [Apollo
+DevTools](https://github.com/apollographql/apollo-client-devtools) for painless
+debugging & full visibility into your store.
+
+<h2 id="start">Quick start</h2>
+
+To get started, install `apollo-link-state` from npm:
+
+```bash
+npm install apollo-link-state --save
+```
+
+The rest of the instructions assume that you already [set up Apollo
+Client](/docs/react/basics/setup.html#installation) in your application. After
+you install the package, you can create your state link by calling
+`withClientState` and passing in a resolver map. A resolver map describes how to
+retrieve and update your local data.
+
+Let's look at an example where we're using a GraphQL mutation to update whether
+our network is connected with a boolean flag:
+
+```js
+import { withClientState } from 'apollo-link-state';
+
+const stateLink = withClientState({
+  Mutation: {
+    networkStatus: (_, { isConnected }, { cache }) => {
+      const data = {
+        networkStatus: { isConnected, __typename: 'NetworkStatus' },
+      };
+      cache.writeData({ data });
+      return data;
+    },
+  },
+});
+```
+
+To hook up your state link to Apollo Client, concatenate it to the other links
+in your Apollo Link chain. Your state link should be one of the last links in
+your chain. Then, pass your link chain to the Apollo Client constructor.
+
+```js
+const client = new ApolloClient({
+  cache: new InMemoryCache(),
+  link: ApolloLink.from([stateLink, new HttpLink()]),
+});
+```
+
+How do we differentiate a request for local data from a request that hits our
+server? In our query or mutation, we specify which fields are client-only with a
+`@client` directive. This tells our network stack to retrieve or update the data
+in the cache with our resolver map that we passed into our state link.
+
+```js
+const UPDATE_NETWORK_STATUS = gql`
+  mutation updateNetworkStatus($isConnected: Boolean) {
+    networkStatus(isConnected: $isConnected) @client {
+      isConnected
+    }
+  }
+`;
+```
+
+To fire off the mutation from your component, bind your mutation to your
+component via your favorite Apollo view layer integration just like you normally
+would. Here's what this would look like for React:
+
+```js
+const WrappedComponent = graphql(UPDATE_NETWORK_STATUS, {
+  props: ({ mutate }) => ({
+    updateNetworkStatus: isConnected => mutate({ variables: { isConnected } }),
+  }),
+})(NetworkStatus);
+```
+
+What if we want to access our network status data from another component? Since
+we don't know whether our `UPDATE_NETWORK_STATUS` mutation will fire before we
+try to access the data, we should guard against undefined values by providing an
+initial state for our query in our resolver map.
+
+```js
+import { withClientState } from 'apollo-link-state';
+
+const stateLink = withClientState({
```

Comment:
I feel like this looks a little too much like wanting to have a second stateLink with different parameters somewhere else, rather than just telling that you need to add a line to the existing one?  Maybe just include something like

```
  Mutation: { /* as above */ }
```